### PR TITLE
[Close Index API] Propagate tasks ids between Freeze, Close and Verify Shard actions

### PR DESCRIPTION
**Pull request by tlrx**
Note: this pull request will be merged in the `close-index-api-refactoring` branch

This pull request changes the Freeze Index and Close Index actions so that these actions always requires a `Task`. The task's id is then propagated from the Freeze action to the Close action, and then to the Verify shard action. This way it is possible to track which Freeze task initiates the closing of an index, and which consecutive verifiy shard are executed for the index closing.

This change has no tests - I could write one using a MockTaskManager listener but it seemed a bit overkill to me.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexClusterStateUpdateRequest.java [hunk 1]
```diff
@@ -25,7 +25,17 @@
  */
 public class CloseIndexClusterStateUpdateRequest extends IndicesClusterStateUpdateRequest<CloseIndexClusterStateUpdateRequest> {
 
+    private long taskId;
+
     public CloseIndexClusterStateUpdateRequest() {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexClusterStateUpdateRequest.java]: make taskId a mandatory parameter by requiring it in the constructor and making the field final?

File: server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexClusterStateUpdateRequest.java [hunk 2]
```diff
@@ -25,7 +25,17 @@
...
+    }
+
+    public Long taskId() {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/admin/indices/close/CloseIndexClusterStateUpdateRequest.java]: return `long`, not `Long`?

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java [hunk 3]
```diff
@@ -338,8 +335,11 @@ private void sendVerifyShardBeforeCloseRequest(final IndexShardRoutingTable shar
             }
             final TransportVerifyShardBeforeCloseAction.ShardRequest shardRequest =
                 new TransportVerifyShardBeforeCloseAction.ShardRequest(shardId);
-            if (timeout != null) {
-                shardRequest.timeout(timeout);
+            if (request.ackTimeout() != null) {
+                shardRequest.timeout(request.ackTimeout());
+            }
+            if (request.taskId() != null) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java]: cannot be null?

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by tlrx** [comment]
@elasticmachine run the default distro tests

**Comment by tlrx** [comment]
@elasticmachine test this please

**Comment by tlrx** [comment]
@elasticmachine test this please

**Comment by tlrx** [comment]
@elasticmachine run gradle build tests 1

**Comment by tlrx** [comment]
Thanks @ywelsch

**Pull request merged**